src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,192 @@
+                        return;
+                    } else {
+                        Expr base = ramp->base;
+                        const Add *add = base.as<Add>();
+                        const IntImm *b = add->b.as<IntImm>();
+                        if (!b) {
+                            // Should we be expecting the index to be in simplified
+                            // canonical form, i.e. expression + IntImm?
+                            debug(4) << "HexagonAlignLoads: add->b is not a constant\n";
+                            debug(4) << "HexagonAlignLoads: Type: " << op->type << "\n";
+                            debug(4) << "HexagonAlignLoads: Index: " << index << "\n";
+                            expr = op;
+                            return;
+                        }
+                        Expr base_low = base - (base_mod + rem_mod);
+                        Expr ramp_low = Ramp::make(simplify(base_low), 1, lanes);
+                        Expr ramp_high = Ramp::make(simplify(base_low + lanes), 1, lanes);
+                        Expr load_low = Load::make(op->type, op->name, ramp_low, op->image, op->param);
+                        Expr load_high = Load::make(op->type, op->name, ramp_high, op->image, op->param);
+                        // slice_vector considers the vectors in it to be concatenated and the result
+                        // is a vector containing as many lanes as the last argument and the vector
+                        // begins at lane number specified by the penultimate arg.
+                        // So, slice_vector(2, a, b, 126, 128) gives a vector of 128 lanes starting
+                        // from lanes 126 through 253 of the concatenated vector a.b.
+                        debug(4) << "HexagonAlignLoads: Unaligned Load: Converting " << (Expr) op << " into ...\n";
+                        expr = Call::make(op->type, Call::slice_vector, { make_two(Int(32)), load_low, load_high, make_const(Int(32), (base_mod + rem_mod)),
+                                    make_const(Int(32), lanes) }, Call::PureIntrinsic);
+                        debug(4) <<  "... " << expr << "\n";
+                        return;
+                    }
+                } else if (ramp && stride && stride->value == 2) {
+                    // We'll try to break this into two dense loads followed by a shuffle.
+                    Expr base_a = ramp->base, base_b = ramp->base + ramp->lanes;
+                    bool shifted_b = false;
+                    int lanes = ramp->lanes;
+
+                    if (op->param.defined()) {
+                        // We need to be able to figure out if buffer_base + base_a is aligned. If not,
+                        // we may have to shift base_b left by one so as to not read beyond the end
+                        // of an external buffer.
+                        ModulusRemainder mod_rem = get_alignment_info(base_a);
```

Comment:
There's a ton of duplicated logic between this case and the previous case. It would be nice to find a way to refactor it so it can be shared. Not required, but please think about if it can be done without adding too much complexity.